http-client/src/test/java/io/airlift/http/client/TestTraceTokenRequestFilter.java

```diff
@@ -43,7 +44,7 @@ public void testBasic()
         assertEquals(filtered.getUri(), original.getUri());
         assertEquals(original.getHeaders().size(), 0);
         assertEquals(filtered.getHeaders().size(), 1);
-        assertEquals(filtered.getHeaders().get(TRACETOKEN_HEADER), ImmutableList.of("testBasic"));
+        assertEquals(filtered.getHeaders().get(TRACETOKEN_HEADER.toLowerCase(ENGLISH)), ImmutableList.of("testBasic"));
```

Comment:
With lowercasing, apps using airlift may need to change too. For example https://github.com/trinodb/trino/blob/a27e5af2d7389a7bc0e906210eafb906f01e4365/core/trino-main/src/test/java/io/trino/operator/MockExchangeRequestProcessor.java#L103

Reply:
We should probably make `Request.getHeaders()` return a case-insensitive map

Reply:
happy to apply once we know for sure (without "probably"). there was quite a bit back and forth already on this PR. It doesn't cost much because it is small, but  it's not cool nonetheless



@electrum @martint can you explicitly decide on the behavior you want to see implemented? happy to apply whatever is needed